internal/plugintest/working_dir.go

```diff
@@ -83,8 +89,20 @@ func (wd *WorkingDir) GetHelper() *Helper {
 // Destroy to establish the configuration. Any previously-set configuration is
 // discarded and any saved plan is cleared.
 func (wd *WorkingDir) SetConfig(cfg string) error {
-	configFilename := filepath.Join(wd.baseDir, ConfigFileName)
-	err := ioutil.WriteFile(configFilename, []byte(cfg), 0700)
+	if wd.configFilename != "" {
+		err := os.Remove(wd.configFilename)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+	}
```

Comment:
Rather than checking for the filename, we should unilaterally remove/overwrite both the HCL and JSON configuration files here, just to prevent any potential oddities. If we want to be slightly more efficient, it can be done based on the `json.Valid()` conditional below 👍 

```go
if json.Valid(bCfg) {
	wd.configFilename = ConfigFileNameJSON

	if err := os.Remove(filepath.Join(wd.baseDir, ConfigFileName)); err != nil && !errors.Is(err, fs.ErrNotExist) {
		return fmt.Errorf("unable to remove %q: %w", ConfigFileName, err)
	}
} else {
	wd.configFilename = ConfigFileName

	if err := os.Remove(filepath.Join(wd.baseDir, ConfigFileNameJSON)); err != nil && !errors.Is(err, fs.ErrNotExist) {
		return fmt.Errorf("unable to remove %q: %w", ConfigFileNameJSON, err)
	}
}
```

Reply:
Done.

I have also added a test that all of the deleting works.

Thanks for review, @bflad
PTAL